### PR TITLE
Add source registry to identify source classes

### PR DESCRIPTION
**Pull request by cescgina**
This new class contains a mapping between driver names and source
classes, so the sources defined in the environment can be correctly
found and constructed.

### CODE REVIEW

File: cibyl/sources/source_registry.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+# pylint: disable=too-few-public-methods
+class SourceRegistry:
+    """Relates driver names to Source classes"""
+    sources = {
+            "jenkins": ("cibyl.sources.jenkins", "Jenkins"),
+            "jenkins_job_builder": ("cibyl.sources.jenkins_job_builder",
+                                    "JenkinsJobBuilder"),
+            "zuul": ("cibyl.sources.zuul", ""),
+            "elk": ("cibyl.sources.elk", "")
```
Comment (adrianfusco) [inline cibyl/sources/source_registry.py:26]: Shouldn't be `elasticsearch` insted of `elk`?

Comment (cescgina) [reply cibyl/sources/source_registry.py:26]: Probably, but since this and the zuul source are still not implemented I just put a placeholder.

**Review by adrianfusco — APPROVED** [review]

**Review by IgnacioPro — APPROVED** [review]
Looks good to me.

**Comment by bregman-arie** [comment]
Good idea

**Comment by bregman-arie** [comment]
Good idea

**Review by bregman-arie — DISMISSED** [review]

**Review by bregman-arie — CHANGES_REQUESTED** [review]
Thinking about it, can we try to use importlib in case the source is not in the registry?

**Comment by cescgina** [comment]
You mean having the previous way as fallback?

**Comment by bregman-arie** [comment]
> You mean having the previous way as fallback?

Exactly

**Review by bregman-arie — APPROVED** [review]

**Pull request merged**